reconcile/utils/acs/policies.py

```diff
@@ -0,0 +1,151 @@
+from typing import Any, Optional
+
+from pydantic import BaseModel
+
+from reconcile.utils.acs.base import AcsBaseApi
+
+
+class Scope(BaseModel):
+    """
+    Scope represents a single cluster or namespace that an acs security policy is applied to.
+    """
+
+    cluster: str
+    namespace: Optional[str]
+
+
+class PolicyCondition(BaseModel):
+    """
+    PolicyCondition represents a single condition criteria enforced within an ACS security policy.
+    Current attributes support subset of "build" lifecycle condition criteria.
+    See section 6.4.3 for table of all condition criteria:
+    https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/4.3/html/operating/manage-security-policies
+    """
+
+    field_name: str
+    negate: Optional[bool]
+    values: list[str]
+
+
+class Policy(BaseModel):
+    """
+    Policy is minimum attributes required to represent an ACS security policy
+    https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/4.3/html/operating/manage-security-policies
+    """
+
+    name: str
+    description: str
+    notifiers: list[str]
+    categories: list[str]
+    severity: str
+    scope: list[Scope]
+    conditions: list[PolicyCondition]
+
+
+class AcsPolicyApi(AcsBaseApi):
+    """
+    Implements methods to support reconcile operations against the ACS PolicyService api
+    """
+
+    def _build_policy(
+        self, api_policy: Any, conditions: list[PolicyCondition]
+    ) -> Policy:
+        return Policy(
+            name=api_policy["name"],
+            description=api_policy["description"],
+            notifiers=sorted(api_policy["notifiers"]),
+            categories=sorted(api_policy["categories"]),
+            severity=api_policy["severity"],
+            scope=sorted(
+                [
+                    Scope(
+                        cluster=s["cluster"],
+                        namespace=s["namespace"],
+                    )
+                    for s in api_policy["scope"]
+                ],
+                key=lambda s: s.cluster,
+            ),
+            conditions=conditions,
+        )
+
+    def _build_policy_condition(self, api_policy_group: Any) -> PolicyCondition:
+        return PolicyCondition(
+            field_name=api_policy_group["fieldName"],
+            values=[v["value"] for v in api_policy_group["values"]],
+            negate=api_policy_group["negate"],
+        )
+
+    def list_custom_policies(self) -> list[Any]:
+        # retrieve summary data for each custom policy
+        return [
+            p
+            for p in self.generic_request("/v1/policies", "GET").json()["policies"]
```

Comment:
Will `/v1/policies` return paginated result?

Reply:
The swagger indicates that there is support for pagination but I've had no luck getting the api to adhere to the parameters with local testing. Here's the swagger json for the route:
```json
"/v1/policies": {
      "get": {
        "summary": "ListPolicies returns the list of policies.",
        "operationId": "PolicyService_ListPolicies",
        "responses": {
          "200": {
            "description": "A successful response.",
            "schema": {
              "$ref": "#/definitions/v1ListPoliciesResponse"
            }
          },
          "default": {
            "description": "An unexpected error response.",
            "schema": {
              "$ref": "#/definitions/runtimeError"
            }
          }
        },
        "parameters": [
          {
            "name": "query",
            "in": "query",
            "required": false,
            "type": "string"
          },
          {
            "name": "pagination.limit",
            "in": "query",
            "required": false,
            "type": "integer",
            "format": "int32"
          },
          {
            "name": "pagination.offset",
            "in": "query",
            "required": false,
            "type": "integer",
            "format": "int32"
          },
          {
            "name": "pagination.sortOption.field",
            "in": "query",
            "required": false,
            "type": "string"
          },
          {
            "name": "pagination.sortOption.reversed",
            "in": "query",
            "required": false,
            "type": "boolean"
          },
          {
            "name": "pagination.sortOption.aggregateBy.aggrFunc",
            "in": "query",
            "required": false,
            "type": "string",
            "enum": [
              "UNSET",
              "COUNT",
              "MIN",
              "MAX"
            ],
            "default": "UNSET"
          },
          {
            "name": "pagination.sortOption.aggregateBy.distinct",
            "in": "query",
            "required": false,
            "type": "boolean"
          }
        ],
        "tags": [
          "PolicyService"
        ]
      }
```

Reply:
and note `"summary": "ListPolicies returns the list of policies."` is a bit misleading as its not the full set of attributes for each policy.

Reply:
We just need to keep it in mind that there will be issues if it returning paginated response back when there are too many policies. `pagination.limit` and `pagination.offset` looks like their way of navigate pages, not sure if unset how many items can return in one response.